src/cargo/ops/cargo_generate_lockfile.rs

```diff
@@ -106,4 +121,23 @@ pub fn update_lockfile(manifest_path: &Path,
             None => {}
         }
     }
+
+    fn compare_dependency_graphs<'a>(previous_resolve: &'a Resolve,
+                                     resolve: &'a Resolve) ->
+                                     Vec<(Option<&'a PackageId>, Option<&'a PackageId>)> {
+        let mut changes = HashMap::new();
+        for dep in previous_resolve.iter() {
+            changes.insert(dep.name(), (Some(dep), None));
```

Comment:
That's a special case because openssl was a system library (so only one is allowed), but in general multiple Rust libraries are indeed allowed.